src/Symfony/Component/HttpFoundation/JsonResponse.php

```diff
@@ -43,6 +43,16 @@ public function __construct($data = null, int $status = 200, array $headers = []
     {
         parent::__construct('', $status, $headers);
 
+        if ($json && !\is_string($data) && !is_numeric($data) && !\is_callable([$data, '__toString'])) {
+            throw new \TypeError(
+                sprintf(
+                    '%s: If $json is set to true, argument $data must be a string or object implementing __toString(), "%s" given.',
+                    __METHOD__,
+                    \get_debug_type($data)
+                )
+            );
```

Comment:
We usually write `throw` statements as a single line. See also the fabbot failure.